src/libLookup/Lookup.cpp

```diff
@@ -1676,25 +1633,28 @@ bool Lookup::InitMining()
     {
         return false;
     }
+
     // Check whether is the new node connected to the network. Else, initiate re-sync process again.
-    this_thread::sleep_for(chrono::seconds(BACKUP_POW2_WINDOW_IN_SECONDS));
-    m_startedPoW2 = false;
-    if (m_syncType != SyncType::NO_SYNC)
+    std::unique_lock<std::mutex> cv_lk(m_mediator.m_node->m_mutexFinishPOW);
+
+    if (m_mediator.m_node->m_cvFinishPOW.wait_for(cv_lk,
+                                                  std::chrono::seconds(90))
+        == std::cv_status::timeout)
     {
         LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                  "Not yet connected to network");
+                  "POW not finished for new node, cannot connected to network");
+
+        return false;
     }
-    else
+
+    m_startedPoW1 = false;
+    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "I have successfully join the network");
+    LOG_GENERAL(INFO, "Clean TxBodyDB except the last one");
+    int size_txBodyDBs = (int)BlockStorage::GetBlockStorage().GetTxBodyDBSize();
+    for (int i = 0; i < size_txBodyDBs - 1; i++)
```

Comment:
If size_txBodyDBs is 0, -1 will return in a integer underflow

Reply:
ok